src/main/java/io/openliberty/tools/intellij/lsp4mp4ij/psi/internal/config/java/MicroProfileConfigASTValidator.java

```diff
@@ -232,10 +237,14 @@ private boolean isAssignable(PsiType fieldBinding, Module javaProject, String de
 	}
 
 	private static boolean doesPropertyHaveValue(String property, JavaDiagnosticsContext context) {
+		PsiMicroProfileProject mpProject = getMicroProfileProject(context);
+		return mpProject.hasProperty(property);
+	}
+
+	private static PsiMicroProfileProject getMicroProfileProject(JavaDiagnosticsContext context) {
```

Comment:
I noticed this method is getting called in a couple new places. Unsure if it's related to this plugin error I'm getting:
```
java.lang.Throwable: Read access is allowed from inside read-action (or EDT) only (see com.intellij.openapi.application.Application.runReadAction())
Current thread: Thread[JobScheduler FJ pool 0/9,4,main] 2088624512 (EventQueue.isDispatchThread()=false)
SystemEventQueueThread: Thread[AWT-EventQueue-0,6,main] 330472837
	at com.intellij.openapi.diagnostic.Logger.error(Logger.java:202)
	at com.intellij.openapi.application.impl.ApplicationImpl.assertReadAccessAllowed(ApplicationImpl.java:1004)
	at com.intellij.workspaceModel.core.fileIndex.impl.WorkspaceFileIndexDataImpl.ensureIsUpToDate(WorkspaceFileIndexDataImpl.kt:127)
	at com.intellij.workspaceModel.core.fileIndex.impl.WorkspaceFileIndexDataImpl.getFileInfo(WorkspaceFileIndexDataImpl.kt:70)
	at com.intellij.workspaceModel.core.fileIndex.impl.WorkspaceFileIndexImpl.getFileInfo(WorkspaceFileIndexImpl.kt:220)
	at com.intellij.workspaceModel.core.fileIndex.impl.WorkspaceFileIndexImpl.findFileSetWithCustomData(WorkspaceFileIndexImpl.kt:205)
	at com.intellij.openapi.roots.impl.ModuleFileIndexImpl.isInTestSourceContent(ModuleFileIndexImpl.java:113)
	at io.openliberty.tools.intellij.lsp4mp4ij.psi.internal.core.ls.PsiUtilsLSImpl.getClasspathKind(PsiUtilsLSImpl.java:190)
	at io.openliberty.tools.intellij.lsp4mp4ij.psi.core.PropertiesManager.getMicroProfileProjectInfo(PropertiesManager.java:86)
	at io.openliberty.tools.intellij.lsp4mp4ij.psi.core.PropertiesManager.getMicroProfileProjectInfo(PropertiesManager.java:78)
	at io.openliberty.tools.intellij.lsp4mp.lsp.MicroProfileLanguageClient.lambda$getProjectInfo$2(MicroProfileLanguageClient.java:119)
	at io.openliberty.tools.intellij.lsp4mp.lsp4ij.LanguageClientImpl.lambda$runAsBackground$4(LanguageClientImpl.java:113)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:186)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$13(CoreProgressManager.java:604)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:679)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:635)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:603)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:60)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:173)
	at io.openliberty.tools.intellij.lsp4mp.lsp4ij.LanguageClientImpl.lambda$runAsBackground$5(LanguageClientImpl.java:111)
	at io.openliberty.tools.intellij.lsp4mp.lsp4ij.LanguageClientImpl.lambda$runAsBackground$6(LanguageClientImpl.java:121)
	at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1804)
	at java.base/java.util.concurrent.CompletableFuture$AsyncRun.exec(CompletableFuture.java:1796)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
```

Reply:
I've seen this plugin error before my changes. This class (`MicroProfileConfigASTValidator`) isn't in the stack trace so not sure what method calls you're referring to.